spinta/manifests/open_api/helpers.py

```diff
@@ -2,99 +2,325 @@
+    ):
+        self.basename: str = basename
+        self.json_schema: dict = json_schema
+        self.name: str = to_property_name(basename)
+        self.title: str = json_schema.get("title", "")
+        self.description: str = json_schema.get("description", "")
+        self.source: str = basename if source is None else source
+        self.datatype: str = datatype or self.get_datatype()
+        self.ref: Model = ref_model
+        self.enum: dict = self.get_enums(self.json_schema.get("enum", []))
+        self.required: bool = not bool(json_schema.get("nullable", False))
+
+
+    def get_datatype(self) -> str:
+        """
+        Returns Property datatype as a string. If type cannot be detected, defaults to "string"
+        """
+        datatype = self.json_schema.get("type", "string")
+
+        basic_types = {
+            "boolean": "boolean",
+            "integer": "integer",
+            "number": "number",
+            "array": "array",
+            "object": "ref",
+        }
+
+        date_time_types = {"date-time": "datetime", "date": "date", "time": "time"}
+
+        if datatype in basic_types:
+            return basic_types[datatype]
+
+        if datatype == "string":
+            if self.json_schema.get("contentEncoding") == "base64":
+                return "binary"
+            if (string_format := self.json_schema.get("format")) in date_time_types:
+                return date_time_types[string_format]
+
+        return "string"
+    
+    def get_enums(self, items:list) -> dict:
+        enum = {}
+        for item in items:
+            if isinstance(item,(list, dict)):  # Only handling primitive type enums
+                return {}
+            enum[item] = {"source": item}
+        return enum
+
+    def get_node_schema_dict(self) -> dict:
+        schema = {
+            "type": self.datatype,
+            "title": self.title,
+            "description": self.description,
+            "external": {"name": self.source},
+            "required": self.required,
+        }
+
+        if self.datatype in ["ref", "backref"]:
+            schema["model"] = self.ref.name
+
+        if self.enum:
+            schema['enums'] = {"": self.enum}
+
+        return schema
+
+
+def get_schema_from_response(response: dict, root: dict) -> dict:
+    json_schema = {}
+    if "openapi" in root:
+        for content_type, content in response.get("content", {}).items():
+            if content_type == "application/json" or content_type.endswith("+json"):
+                json_schema = resolve_refs(content.get("schema", {}), root)
+                break
+    elif "swagger" in root:
+        json_schema = resolve_refs(response.get("schema", {}), root)
+    else:
+        raise ValueError("Unknown specification type. Only OpenAPI 3.* and Swagger 2.0 allowed.")
+
+    return json_schema
+
+
+def get_model_schemas(dataset_name: str, resource_name: str, response_200: dict, root: dict) -> list[dict]:
```

Comment:
The others should not even come here (unless 201 is also an option), but still, I'd keep it more generalised.

```suggestion
def get_model_schemas(dataset_name: str, resource_name: str, response: dict, root: dict) -> list[dict]:
```